### PR TITLE
Adds a reagent dart gun for crew

### PR DESCRIPTION
**Pull request by thelaughingbomb**
## About The Pull Request

It adds a reskinned (given RGB value) reagent dart gun that crew can research and build. Its locked behind advanced biotech technology. Also the gun can only be loaded by medical chems.

## Why It's Good For The Game

Rapid deployment of medicine is good.

## Changelog
:cl:
add: Medical dart gun to research and build at medical lathe
/:cl:

### CODE REVIEW

**Comment by TheFakeElon** [comment]
don't allow pax

File: austation/code/modules/projectiles/guns/misc/chem_gun.dm [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+
+/obj/item/gun/chem/medical/attackby(obj/item/I, mob/user)
+	..()
+	if(istype(I, /obj/item/reagent_containers))
+		for(var/datum/reagent/R in I.reagents.reagent_list)
+			if(R.type in allowedchems)
+				return FALSE
+			else
+				to_chat(user, "<span class='notice'>The dart gun refuses as [I] has harmful chemicals.</span>")
+				return TRUE
```
Comment (MCterra10) [inline austation/code/modules/projectiles/guns/misc/chem_gun.dm:64]: doesn't this call the transfer code from the syndi chem gun BEFORE the actual check, transferring anyway?

Comment (thelaughingbomb) [reply austation/code/modules/projectiles/guns/misc/chem_gun.dm:64]: play testing shows that it actually rejects the chemicals

Comment (TheFakeElon) [inline austation/code/modules/projectiles/guns/misc/chem_gun.dm:64]:
```suggestion
	if(istype(I, /obj/item/reagent_containers))
		for(var/datum/reagent/R in I.reagents.reagent_list)
			if(R.type in allowedchems)
				return FALSE
			
		to_chat(user, "<span class='notice'>The dart gun refuses as [I] has harmful chemicals.</span>")
		return TRUE

```

This should properly stop poisons from entering the gun

Comment (TheFakeElon) [inline austation/code/modules/projectiles/guns/misc/chem_gun.dm:64]:
```suggestion
/obj/item/gun/chem/medical/attackby(obj/item/I, mob/user)
	..()
	if(istype(I, /obj/item/reagent_containers))
		for(var/datum/reagent/R in I.reagents.reagent_list)
			if(!R.type in allowedchems)
				return TRUE
				to_chat(user, "<span class='notice'>The dart gun refuses as [I] has harmful chemicals.</span>")
		
		to_chat(user, "<span class='notice'>The dart gun refuses as [I] has harmful chemicals.</span>")
		return FALSE

```

**Comment by thelaughingbomb** [comment]
Its fixed now

File: austation/code/modules/projectiles/guns/misc/chem_gun.dm [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+// chem check
+
+/obj/item/gun/chem/medical/attackby(obj/item/I, mob/user)
+	..()
+	if(istype(I, /obj/item/reagent_containers))
+		for(var/datum/reagent/R in I.reagents.reagent_list)
+			if(R.type in allowedchems)
+
+			else
```
Comment (TheFakeElon) [inline austation/code/modules/projectiles/guns/misc/chem_gun.dm:62]: The shit code levels are off the charts lmao, change it to if(!R.type

Comment (LemonInTheDark) [reply austation/code/modules/projectiles/guns/misc/chem_gun.dm:62]: wait that compiles?

Comment (thelaughingbomb) [reply austation/code/modules/projectiles/guns/misc/chem_gun.dm:62]: it just works

Comment (TheFakeElon) [reply austation/code/modules/projectiles/guns/misc/chem_gun.dm:62]: Make it stop make it stop make it stop

File: austation/code/modules/projectiles/guns/misc/chem_gun.dm [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	color = rgb(0,255,255) //cyan
+	force = 4
+	materials = list(/datum/material/iron=2000)
+	clumsy_check = FALSE
+	fire_sound = 'sound/items/syringeproj.ogg'
+	fire_rate = 1
+	time_per_syringe = 200
+	syringes_left = 6
+	max_syringes = 6
+	last_synth = 0
```
Comment (park66665) [inline austation/code/modules/projectiles/guns/misc/chem_gun.dm:18]:
```suggestion
	color = rgb(0,255,255) //cyan
	fire_rate = 1
	time_per_syringe = 200
	syringes_left = 6
	max_syringes = 6
```
Please consider removing redundant lines.

**Review by park66665 — CHANGES_REQUESTED** [review]

File: austation/code/modules/projectiles/guns/misc/chem_gun.dm [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+/obj/item/gun/chem/medical/attackby(obj/item/I, mob/user)
+	..()
+	if(istype(I, /obj/item/reagent_containers))
+		for(var/datum/reagent/R in I.reagents.reagent_list)
+			if(R.type in allowedchems)
+
+			else
+				to_chat(user, "<span class='notice'>The dart gun refuses as [I] has harmful chemicals.</span>")
+				return TRUE
```
Comment (TheFakeElon) [inline austation/code/modules/projectiles/guns/misc/chem_gun.dm:28]:
```suggestion
	if(istype(I, /obj/item/reagent_containers))
		for(var/datum/reagent/R in I.reagents.reagent_list)
			if(!(R.type in allowedchems))
				to_chat(user, "<span class='notice'>The dart gun refuses as [I] has harmful chemicals.</span>")
				return TRUE
```

**Review by TheFakeElon — CHANGES_REQUESTED** [review]

File: austation/code/modules/projectiles/guns/misc/chem_gun.dm [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/obj/item/gun/chem/medical
+	name = "medical darter"
```
Comment (TheFakeElon) [inline austation/code/modules/projectiles/guns/misc/chem_gun.dm:2]: Dart gun would sound better

File: austation/code/modules/projectiles/guns/misc/chem_gun.dm [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+	fire_rate = 1
+	time_per_syringe = 200
+	syringes_left = 6
+	max_syringes = 6
+	casing = /obj/item/ammo_casing/chemgun/medical
+	var/list/allowedchems
+
+/obj/item/gun/chem/medical/Initialize()
+	. = ..()
+	allowedchems = typesof(/datum/reagent/medicine) + typesof(/datum/reagent/vaccine) + typesof(/datum/reagent/water) //easier to define allowed chems then to blacklist them
```
Comment (TheFakeElon) [inline austation/code/modules/projectiles/guns/misc/chem_gun.dm:14]: Hydration is important

**Review by TheFakeElon — APPROVED** [review]

File: austation/code/modules/projectiles/guns/misc/chem_gun.dm (hunk 6 above)
Comment (thelaughingbomb) [reply austation/code/modules/projectiles/guns/misc/chem_gun.dm:14]: yes

File: austation/code/modules/projectiles/guns/misc/chem_gun.dm (hunk 5 above)
Comment (thelaughingbomb) [reply austation/code/modules/projectiles/guns/misc/chem_gun.dm:2]: Dart pistol exists, and having a dart gun and dart pistol is going to confuse people on comms.

File: austation/code/modules/projectiles/guns/misc/chem_gun.dm (hunk 6 above)
Comment (thelaughingbomb) [reply austation/code/modules/projectiles/guns/misc/chem_gun.dm:14]: yes

File: austation/code/modules/projectiles/guns/misc/chem_gun.dm (hunk 5 above)
Comment (TheFakeElon) [reply austation/code/modules/projectiles/guns/misc/chem_gun.dm:2]: I don't think there's such thing as a medical dart gun, unless you are referring to the syringe gun

**Comment by TheFakeElon** [comment]
okay fine

**Pull request merged**